internal/app/mapper/mapper.go

```diff
@@ -116,6 +136,42 @@ func isAnyMatch(matchPredicate *matchPredicate) bool {
 	return matchPredicate.GetAnyMatch()
 }
 
+func isAndMatch(matchPredicate *matchPredicate, typeURL string, node *core.Node) (bool, error) {
+	matchset := matchPredicate.GetAndMatch()
+	if matchset == nil {
+		return false, nil
+	}
+
+	for _, rule := range matchset.GetRules() {
+		isMatch, err := isMatch(rule, typeURL, node)
+		if err != nil {
+			return false, err
+		}
+		if !isMatch {
+			return false, nil
+		}
+	}
```

Comment:
This is very clean, I liked how you handled both these predicate scenarios.

Reply:
done